src/open_dread_rando/pickups/split_pickups.py

```diff
@@ -28,29 +33,28 @@ class MissileInput(Enum):
     L = 1
 
 
+T = TypeVar('T', bound=ArgAnyType)
+
+
 class CCFunc:
-    def __init__(self, raw: dict) -> None:
+    def __init__(self, raw: ActorDefFunc) -> None:
         self.raw = raw
 
-    def _param(self, i: int) -> Container:
-        return self.raw["params"][f"Param{i}"]
+    def get_param(self, i: int) -> ArgAnyType:
+        return self.raw.get_param(i)
 
-    def get_param(self, i: int) -> Any:
-        return self._param(i).value
+    def set_param(self, i: int, value: ArgAnyType) -> None:
+        self.raw.set_param(i, value)
 
-    def set_param(self, i: int, value: Any):
-        self._param(i).value = value
 
-
-T = TypeVar('T')
 class Param(Generic[T]):
     def __init__(self, index: int) -> None:
         self.index = index
 
-    def __get__(self, inst: CCFunc, objtype=None) -> T:
```

Comment:
...and this class replaced with ActorDefFuncParam

Reply:
This one does automatically cast things to T so it's helpful